internal/versions/versions.go

```diff
@@ -109,17 +129,95 @@ func (h *handler) GetRHCOSVersion(openshiftVersion string) (string, error) {
 }
 
 func (h *handler) IsOpenshiftVersionSupported(openshiftVersion string) bool {
+	openshiftVersion, err := h.GetSupportedKeyVersionFormat(openshiftVersion)
+	if err != nil {
+		return false
+	}
 	if _, ok := h.openshiftVersions[openshiftVersion]; !ok {
 		return false
 	}
 
 	return true
 }
 
-func (h *handler) GetSupportedVersionFormat(openshiftVersion string) (string, error) {
+// Should return release version (as fetched from 'oc adm release info')
+func (h *handler) GetReleaseVersion(openshiftVersion string) (string, error) {
+	openshiftVersion, err := h.GetSupportedKeyVersionFormat(openshiftVersion)
+	if err != nil {
+		return "", err
+	}
+	if !h.IsOpenshiftVersionSupported(openshiftVersion) {
+		return "", errors.Errorf("No release version for unsupported openshift version %s", openshiftVersion)
+	}
+
+	if h.openshiftVersions[openshiftVersion].ReleaseVersion == nil {
+		return "", errors.Errorf("Release version was missing for openshift version %s", openshiftVersion)
+	}
```

Comment:
This could happen for the default versions that weren't created by `ImageSet` since we don't have a validation anymore for versions having ReleaseVersion.
It would cause a different behavior for KubeAPI and REST API (as `GetReleaseVersion` is used in `RegisterClusterInternal`).

I think that we should verify that `release_version` is set in case `ENABLE_KUBE_API` is false.